### PR TITLE
Generalized checks

### PR DESCRIPTION
**Pull request by vprevosto**
Introduce the new `check` and (future) `admit` clause modifier keywords in the manual.

### CODE REVIEW

**Comment by vprevosto** [comment]
@AllanBlanchard for information.

File: speclang_modern.tex [hunk 1]
```diff
@@ -1060,6 +1060,30 @@ \subsection{Simple function contracts}
...
+\end{itemize}
+Confusion can arise because the words \emph{assigns} and \emph{modifies} are sometimes used interchangeably. In particular, \textbf{ACSL \ifCPP{and \NAME} \iftoggle{isCPP}{use}{uses}
+	\emph{modifies} semantics, even though the frame condition is introduced by the \emph{assigns} keyword}.\footnote{For comparison, JML and the OpenJML tool define frame conditions to have write semantics but use the keywords assigns and modifies interchangeably; however, the KeY tool for JML implements modifies semantics. Ada/SPARK's data flow contracts effectively encode write semantics.}
+
+\subsubsection{\texorpdfstring{\lstinline|check|}{check} and \texorpdfstring{\lstinline|admit|}{admit} clauses}\label{sec:check-admit-clauses}
+
+As presented above, in their basic acceptions, \lstinline|requires|
+and \lstinline|ensures| clauses both must be checked when control reaches the
+corresponding point, and can be assumed to hold to continue the analysis. However
+if only one of these actions, a {\it clause-kind} modifier can be used.
```
Comment (pbaudin) [inline speclang_modern.tex:1080]: About *However... used*, this sentence seems incomprehensible to me.

Comment (AllanBlanchard) [reply speclang_modern.tex:1080]:
```suggestion
if only one of these actions is needed, a {\it clause-kind} modifier can be used.
```

**Comment by vprevosto** [comment]
@pbaudin if you're fine with this latest version, I'll let you do the merge.

**Pull request merged**